### PR TITLE
Expose metrics reporters constructors to end-user

### PR DESCRIPTION
**Pull request by Ardagan**
**What changed?**
Cleaned up and exposed metric reporters constructors to the user.

**Why?**
Simplify reporter construction for the user.

**How did you test it?**
manual/auto tests

**Potential risks**
Metrics reporters are not initialized correctly.

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/metrics/config.go [hunk 1]
```diff
@@ -216,23 +216,26 @@ var (
 // customReporter Provide this argument if you want to report metrics to a custom metric platform, otherwise use nil.
 //
 // returns SeverReporter, SDKReporter, error
-func (c *Config) InitMetricReporters(logger log.Logger, customReporter interface{}) (Reporter, Reporter, error) {
-	// init from config only
-	if customReporter == nil {
-		if c.Prometheus != nil && len(c.Prometheus.Framework) > 0 {
-			return c.initReportersFromPrometheusConfig(logger, customReporter)
-		}
+func InitMetricReporters(logger log.Logger, c *Config, customReporter interface{}) (Reporter, Reporter, error) {
```
Comment (yiminc) [inline common/metrics/config.go:219]: Can we not expose this method? We don't want user to use this method.

File: common/metrics/config.go [hunk 2]
```diff
@@ -216,23 +216,26 @@ var (
...
+	if customReporter != nil {
+		return initMetricsReportersFromCustomReporter(logger, c, customReporter)
+	}
 
-		scope := c.NewScope(logger)
-		reporter := newTallyReporter(scope, &c.ClientConfig)
-		return reporter, reporter, nil
+	// TODO deprecated path support, remove once we deprecate custom SDK reporter (target version 1.17)
+	if c.PrometheusSDK != nil {
```
Comment (yiminc) [inline common/metrics/config.go:225]: why we cannot remove this now?

File: common/metrics/config.go (hunk 1 above)
Comment (Ardagan) [reply common/metrics/config.go:219]: We have to expose it to be accessible by other server packages.

File: common/metrics/config.go [hunk 3]
```diff
@@ -207,96 +205,109 @@ var (
...
+// serverReporter, err := c.InitMetricsReporterInternal(logger, config, customReporter)
 // metricsClient := serverReporter.newClient(logger, serviceIdx)
 //
 // customReporter Provide this argument if you want to report metrics to a custom metric platform, otherwise use nil.
 //
-// returns SeverReporter, SDKReporter, error
-func (c *Config) InitMetricReporters(logger log.Logger, customReporter interface{}) (Reporter, Reporter, error) {
-	// init from config only
+// returns SeverReporter, error
+func InitMetricsReporterInternal(logger log.Logger, c *Config, customReporter interface{}) (Reporter, error) {
```
Comment (yiminc) [inline common/metrics/config.go:217]: do we have to expose this method? the name is internal.

**Review by yiminc — APPROVED** [review]

**Pull request merged**